tests/test_util.py

```diff
@@ -1510,6 +1511,29 @@ def test_introspection_signal_handler(capfd):
     assert log_msg in err
 
 
+@pytest.mark.parametrize('dockerignore_exists', [True, False])
+def test_allow_repo_dir_in_dockerignore(tmpdir, dockerignore_exists):
+    docker_ignore_file = os.path.join(str(tmpdir), DOCKERIGNORE)
+
+    ignore_content = ["# Ignore everything so we can just have a whitelist of things to copy\n",
+                      "** / *\n"]
+    added_lines = ["\n", "!%s\n" % RELATIVE_REPOS_PATH]
```

Comment:
Maybe don't include "\n" in `added_lines`. We don't care whether we get an additional newline... we only care that "!..." appears as its own line.

Reply:
ok